src/metax_api/models/catalog_record.py

```diff
@@ -2015,12 +2010,21 @@ def calculate_directory_byte_sizes_and_file_counts(self):
         """
         Calculate directory byte_sizes and file_counts for all dirs selected for this cr.
         """
-        if not self.research_dataset.get('directories', None):
-            return
-
         _logger.info('Calculating directory byte_sizes and file_counts...')
 
-        dir_identifiers = [ d['identifier'] for d in self.research_dataset['directories'] ]
+        dir_identifiers = file_dir_identifiers = []
+
+        if self.research_dataset.get('directories', None):
+            dir_identifiers = [d['identifier'] for d in self.research_dataset['directories']]
+
+        if self.research_dataset.get('files', None):
+            file_dir_identifiers = [File.objects.get(identifier=f['identifier']).parent_directory.identifier
+                for f in self.research_dataset['files']]
+
```

Comment:
As this files are on top level and not part of directories JSON (research_dataset), those file sizes  are not counted in if not  
separately picked in.